pkg/kubelet/volume/volume_manager.go

```diff
@@ -209,16 +213,42 @@ func (vm *volumeManager) GetVolumesForPodAndAppendSupplementalGroups(
 }
 
 func (vm *volumeManager) GetVolumesInUse() []api.UniqueVolumeName {
-	attachedVolumes := vm.actualStateOfWorld.GetAttachedVolumes()
-	volumesInUse :=
-		make([]api.UniqueVolumeName, 0 /* len */, len(attachedVolumes) /* cap */)
-	for _, attachedVolume := range attachedVolumes {
-		if attachedVolume.PluginIsAttachable {
-			volumesInUse = append(volumesInUse, attachedVolume.VolumeName)
+	// Report volumes in desired state of world and actual state of world so
+	// that volumes are marked in use as soon as the decision is made that the
+	// volume *should* be attached to this node until it is safely unmounted.
+	desiredVolumes := vm.desiredStateOfWorld.GetVolumesToMount()
```

Comment:
Just to clarify, this is populated when a new pod is handed to the pod manager right? how does it work on kubelet restart where we lose internal state, will there be some time where we update the volumes in use to an empty list, and then add volumes as pods are handed to us?